### PR TITLE
Feature:  mcq in assessment

### PR DESCRIPTION
**Pull request by somnath77**

### CODE REVIEW

File: db/migrate/20220127053222_add_problem_count_in_assessments.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class AddProblemCountInAssessments < ActiveRecord::Migration[6.1]
+  def up
```
Comment (pooja-mane) [inline db/migrate/20220127053222_add_problem_count_in_assessments.rb:4]: Please add `down` method in migration

File: app/controllers/api/v1/admin/assessments_controller.rb [hunk 2]
```diff
@@ -142,7 +255,10 @@ def load_assessment
         end
 
         def assessment_params
-          params.permit(:name, :description, :problem_count, :organization_id, :is_sparkode_request, assessments_contacts_attributes: %i[id user_id _destroy])
+          params.permit(:name, :description, :organization_id, :problem_tag_list, :mcq_tag_list,
+                        :is_sparkode_request, problem_count_hash: {}, mcq_count_hash: {},
+                                              problem_tag_list: [], mcq_tag_list: [],
```
Comment (pooja-mane) [inline app/controllers/api/v1/admin/assessments_controller.rb:260]: Why we require `problem_tag_list `, `mcq_tag_list` separately `:problem_tag_list, :mcq_tag_list, problem_tag_list: [], mcq_tag_list: []`

File: db/migrate/20220127053222_add_problem_count_in_assessments.rb (hunk 1 above)
Comment (pooja-mane) [reply db/migrate/20220127053222_add_problem_count_in_assessments.rb:4]:
```
def down
    add_column :assessments, :problem_count

    Assessment.all.each do |assessment|
      assessment.update(problem_count: assessment.problem_count_hash.values.map(&:to_i).sum)
    end

    remove_column :assessments, :problem_count_hash
end
```

Comment (Devyani606) [reply db/migrate/20220127053222_add_problem_count_in_assessments.rb:4]: done

**Review by pooja-mane — APPROVED** [review]

**Review by pooja-mane — APPROVED** [review]

**Review by Devyani606 — APPROVED** [review]

**Pull request merged**